### PR TITLE
[AM][OPS-10184][OPS-10220][OPS-10221] add legacy calculator to legacy calculator feature

### PR DESCRIPTION
**Pull request by almorcrette**
- add legacy calculator and associated `TaxPaymentPlan` model, and its test suites
- connect legacy calculator to three pages displaying payment plans ('How much can you pay each month?', 'Check plan', 'View plan')
- update legacy calculator interface with calculator controller so that the right number of plans is show in edge cases
- Update plan option ids from 50%, 60%, 80% to abstracted `PaymentPlanOption`
- Add tests for legacy pages
- Add feature flag to test config on existing pages where necessary

### CODE REVIEW

File: test/pagespecs/legacycalculator/HowMuchCanYouPayEachMonthPageLegacyCalculatorSpec.scala [hunk 1]
```diff
@@ -76,76 +77,81 @@ class HowMuchCanYouPayEachMonthPageLegacyCalculatorSpec extends ItSpec {
...
-        howMuchCanYouPayEachMonthPage.optionIsNotDisplayed("5,700")
-        howMuchCanYouPayEachMonthPage.optionIsNotDisplayed("7,600")
+    "if a two-month plan is the closest monthly amount less than 50% of remaining income after spending, show only this option" in {
+      beginJourney(4900)
+
+      howMuchCanYouPayEachMonthPageLegacyCalculator.optionIsDisplayed("2,450")
+      howMuchCanYouPayEachMonthPageLegacyCalculator.optionIsNotDisplayed("4,900")
+      howMuchCanYouPayEachMonthPageLegacyCalculator.optionIsNotDisplayed("1,633.33")
+      howMuchCanYouPayEachMonthPageLegacyCalculator.optionIsNotDisplayed("1,633.34")
+      //        howMuchCanYouPayEachMonthPageLegacyCalculator.assertContentDoesNotContainOrSeparator
```
Comment (AndyHWChung) [inline test/pagespecs/legacycalculator/HowMuchCanYouPayEachMonthPageLegacyCalculatorSpec.scala]: this could be cleaned up if it's not being used

File: test/pagespecs/legacycalculator/HowMuchCanYouPayEachMonthPageLegacyCalculatorSpec.scala [hunk 2]
```diff
@@ -76,76 +77,81 @@ class HowMuchCanYouPayEachMonthPageLegacyCalculatorSpec extends ItSpec {
...
+    }
+    "if a three-month plan is the closest monthly amount less than 50% of remaining income after spending, " +
+      "show three-month and two-month plans only" in {
+        beginJourney(3267)
+
+        howMuchCanYouPayEachMonthPageLegacyCalculator.optionIsDisplayed("1,633.33")
+        howMuchCanYouPayEachMonthPageLegacyCalculator.optionIsDisplayed("2,450")
+        howMuchCanYouPayEachMonthPageLegacyCalculator.optionIsNotDisplayed("4,900")
+        howMuchCanYouPayEachMonthPageLegacyCalculator.optionIsNotDisplayed("1,225")
+        //        howMuchCanYouPayEachMonthPageLegacyCalculator.assertContentDoesNotContainOrSeparator
```
Comment (AndyHWChung) [inline test/pagespecs/legacycalculator/HowMuchCanYouPayEachMonthPageLegacyCalculatorSpec.scala]: this could also be cleaned up

File: app/ssttpcalculator/model/PaymentPlanOption.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+sealed trait PaymentPlanOption extends EnumEntry with EnumEntry.Uncapitalised {
+  val idKey: String
+}
+
+object PaymentPlanOption extends Enum[PaymentPlanOption] with PlayInsensitiveJsonEnum[PaymentPlanOption] {
+  val values: immutable.IndexedSeq[PaymentPlanOption] = findValues
+
+  case object Basic extends PaymentPlanOption {
+    val idKey = "basic"
```
Comment (AndyHWChung) [inline app/ssttpcalculator/model/PaymentPlanOption.scala]: I think these `idKey` values are the same ones that you get with `.entryName` from `EnumEntry` - is it possible to use `entryName` instead of `idKey` if it is? Or did you specifically want something called `idKey`? If you did then it might be an option to have:
```scala
sealed trait PaymentPlanOption extends EnumEntry with EnumEntry.Uncapitalised {
  val idKey: String = entryName
}
```
to save having to define the values in the case objects.

**Review by AndyHWChung — APPROVED** [review]
